extension-manifest-v3/src/background/utils/engines.js

```diff
@@ -0,0 +1,336 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import * as IDB from 'idb';
+import {
+  FiltersEngine,
+  ENGINE_VERSION,
+  getLinesWithFilters,
+  mergeDiffs,
+} from '@cliqz/adblocker';
+
+import { registerDatabase } from '/utils/indexeddb.js';
+
+const engines = new Map();
+
+function loadFromMemory(name) {
+  return engines.get(name);
+}
+
+function saveToMemory(name, engine) {
+  engines.set(name, engine);
+}
+
+const DB_NAME = registerDatabase('engines');
+
+async function getDB() {
```

Comment:
why not share with daily-stats?

Reply:
The common code is not that huge, as each db has its own upgrade and blocking implementation. Only the shell of the function is similar. I think it is not worth it.